src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -299,7 +299,8 @@ def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights):
     :return: xgb.DMatrix or None
     """
     try:
-        dataset = [mlio.SageMakerPipe(pipe_path)]
```

Comment:
The existing code (and the refactored code) staring here is not easy to follow with all the nested if statements. Were there any tests for this function? I can't seem to find any tests that test this function.

Reply:
We have test in integration tests.

Reply:
On autopilot side I assume? I would feel more comfortable if we have some tests on the xgboost side since there is significant code change.

Reply:
DMatrix is an one-way door. The only way to combine two input channels, it is to read them together. In fact, mlio lib support to read two pipe channels together. We extend this function a little to leverage that capability.

I add the test in xgb's integration tests. I also did end-to-end tests with local runs and AP.

Because cross validation relies on this capability, I have to make the PR for both.

Reply:
The test is in test_data_utils.py. Scroll down to the bottom.